cmd/faucet/faucet.go

```diff
@@ -613,24 +618,22 @@ func (f *faucet) loop() {
 				continue
 			}
 			// Faucet state retrieved, update locally and send to clients
-			f.lock.RLock()
 			log.Info("Updated faucet state", "number", head.Number, "hash", head.Hash(), "age", common.PrettyAge(timestamp), "balance", f.balance, "nonce", f.nonce, "price", f.price)
-
-			balance := new(big.Int).Div(f.balance, ether)
-			peers := f.stack.Server().PeerCount()
-
+			data, _ := json.Marshal(map[string]interface{}{
+				"funds":    new(big.Int).Div(f.balance, ether),
+				"funded":   f.nonce,
```

Comment:
Looks like the `f.nonce`, `f.reqs`, etc internal states need the lock protection?